examples/platform/silabs/BaseApplication.cpp

```diff
@@ -251,7 +251,11 @@ void BaseApplicationDelegate::OnCommissioningWindowClosed()
         SilabsLCD::Screen_e screen;
         slLCD.GetScreen(screen);
         VerifyOrReturn(screen == SilabsLCD::Screen_e::QRCodeScreen);
-        slLCD.SetScreen(SilabsLCD::Screen_e::DemoScreen);
+        AppEvent event;
+        event.Type            = AppEvent::kEventType_LCD;
+        event.LCDEvent.screen = reinterpret_cast<void *>(static_cast<uintptr_t>(SilabsLCD::Screen_e::DemoScreen));
```

Comment:
Why don't we use the Screen_e enum type here rather than all those casting?

Reply:
done